src/main/java/de/unistuttgart/iste/gits/media_service/controller/MediaController.java

```
@@ -25,63 +37,55 @@ public class MediaController {
-                ids,
-                uploadUrlInSelectionSet(env),
-                downloadUrlInSelectionSet(env)
-        );
+    public List<MediaRecord> mediaRecordsByIds(@Argument final List<UUID> ids, final DataFetchingEnvironment env, @ContextValue final LoggedInUser currentUser) {
+        final List<MediaRecord> mediaRecords = mediaService.getMediaRecordsByIds(ids, uploadUrlInSelectionSet(env), downloadUrlInSelectionSet(env));
+
+        return checkAccessForMediaRecords(currentUser, mediaRecords, UserRoleInCourse.STUDENT);
     }
 
     @QueryMapping
-    public List<MediaRecord> findMediaRecordsByIds(@Argument final List<UUID> ids, final DataFetchingEnvironment env) {
-        return mediaService.findMediaRecordsByIds(
-                ids,
-                uploadUrlInSelectionSet(env),
-                downloadUrlInSelectionSet(env)
-        );
+    public List<MediaRecord> findMediaRecordsByIds(@Argument final List<UUID> ids, final DataFetchingEnvironment env, @ContextValue final LoggedInUser currentUser) {
+        final List<MediaRecord> mediaRecords = mediaService.findMediaRecordsByIds(ids, uploadUrlInSelectionSet(env), downloadUrlInSelectionSet(env));
+
+        return checkAccessForMediaRecords(currentUser, mediaRecords, UserRoleInCourse.STUDENT);
     }
 
+
     @QueryMapping
     public List<MediaRecord> userMediaRecords(@ContextValue final LoggedInUser currentUser,
                                               final DataFetchingEnvironment env) {
-        return mediaService.getMediaRecordsForUser(
-                currentUser.getId(),
-                uploadUrlInSelectionSet(env),
-                downloadUrlInSelectionSet(env)
-        );
+        final List<MediaRecord> mediaRecords = mediaService.getMediaRecordsForUser(currentUser.getId(), uploadUrlInSelectionSet(env), downloadUrlInSelectionSet(env));
+
+        return checkAccessForMediaRecords(currentUser, mediaRecords, UserRoleInCourse.STUDENT);
```

should throw an error, not return null